pkg/plugins/resources/yaml/target.go

```diff
@@ -60,55 +66,89 @@ func (y *Yaml) target(source string, dryRun bool) (bool, []string, string, error
-		logrus.Infof("%s Key '%s', from file '%v', already set to %s, nothing else need to be done",
-			result.SUCCESS,
-			y.spec.Key,
-			y.spec.File,
-			valueToWrite)
-		return false, files, message, nil
-	}
-	logrus.Infof("%s Key '%s', from file '%v', was updated from '%s' to '%s'",
-		result.ATTENTION,
-		y.spec.Key,
-		y.spec.File,
-		oldVersion,
-		valueToWrite)
+		out := yaml.Node{}
 
-	if !dryRun {
+		err := yaml.Unmarshal([]byte(y.files[filePath]), &out)
 
-		newFile, err := os.Create(y.spec.File)
+		if err != nil {
+			return false, files, message.String(), fmt.Errorf("cannot unmarshal content of file %s: %v", filePath, err)
+		}
 
-		// https://staticcheck.io/docs/checks/#SA5001
-		//lint:ignore SA5001 We want to defer the file closing before exiting the function
-		defer newFile.Close()
+		keyFound, oldVersion, _ := replace(&out, strings.Split(y.spec.Key, "."), valueToWrite, 1)
 
-		if err != nil {
-			return false, files, message, nil
+		if !keyFound {
+			return false, files, message.String(), fmt.Errorf("%s cannot find key '%s' from file '%s'",
+				result.FAILURE,
+				y.spec.Key,
+				filePath)
 		}
 
-		encoder := yaml.NewEncoder(newFile)
-		defer encoder.Close()
-		encoder.SetIndent(yamlIdent)
-		err = encoder.Encode(&out)
+		if oldVersion == valueToWrite {
+			logrus.Infof("%s Key '%s', from file '%v', already set to %s, nothing else need to be done",
+				result.SUCCESS,
+				y.spec.Key,
+				filePath,
+				valueToWrite)
+			notChanged++
+		} else {
+			newFileContent, err := yaml.Marshal(&out)
+			if err != nil {
+				return false, files, message.String(), err
+			}
+			y.files[filePath] = string(newFileContent)
+
+			files = append(files, filePath)
+
+			logrus.Infof("%s Key '%s', from file '%v', was updated from '%s' to '%s'",
+				result.ATTENTION,
+				y.spec.Key,
+				filePath,
+				oldVersion,
+				valueToWrite)
+		}
 
-		if err != nil {
-			return false, files, message, err
+		if !dryRun {
+
+			newFile, err := os.Create(filePath)
+
+			// https://staticcheck.io/docs/checks/#SA5001
+			//lint:ignore SA5001 We want to defer the file closing before exiting the function
+			defer newFile.Close()
+
+			if err != nil {
+				return false, files, message.String(), nil
+			}
+
+			encoder := yaml.NewEncoder(newFile)
+			defer encoder.Close()
+			encoder.SetIndent(yamlIndent)
+			err = encoder.Encode(&out)
+			if err != nil {
+				return false, files, message.String(), err
+			}
+			// TODO: check if all this encoder stuff is needed, and why yamlIndent isn't respected
+			err = y.contentRetriever.WriteToFile(y.files[filePath], filePath)
+			if err != nil {
+				return false, files, message.String(), err
+			}
 		}
+		message.WriteString(fmt.Sprintf("Update key %q from file %q", y.spec.Key, filePath))
+
 	}
 
-	files = append(files, y.spec.File)
+	// If no file was updated, return an error
+	// TODO: why?
+	if notChanged == len(y.files) {
+		return false, files, message.String(), fmt.Errorf("no file was updated")
```

Comment:
Why are you returning an error if nothing changed? 
Wouldn't it be better to return something like
```suggestion
		return false, files, message.String(), nil
```

Reply:
https://github.com/updatecli/updatecli/pull/650/commits/4de4f95c7b4509f1ce1377b4ac2efb1d212f7ec4